### PR TITLE
Floppy software additions (Amiga, ST, IBM PC, Mac)

### PR DESCRIPTION
**Pull request by FakeShemp**
* Add a bunch of new dumps
* Redumped and/or reconverted dumps that had various issues
* Tested a bunch of dumps to add _supported_ status and/or _baddump_
* Add more info about various dumps

### CODE REVIEW

**Comment by ajrhacker** [comment]
As MAMEdevs have been discussing recently on a private forum, it's important for contributors of new software dumps to list their sources. Please do so.

**Comment by FakeShemp** [comment]
@ajrhacker List their sources in the softlist itself or here? Pretty much all these dumps are made by me using a Kryoflux.

**Comment by ajrhacker** [comment]
Putting it in the commit message is best. It's needed for compiling release notes (as is being done right now for MAME 0.213).

**Comment by FakeShemp** [comment]
@ajrhacker No links or anything, right?

**Comment by ajrhacker** [comment]
Links are not necessary. All that's needed is some kind of identifier.

You should be all right here, but other contributors have been adding variously-sourced software dumps in very large batches and tracking them down has been a nuisance.

File: hash/ibm5170.xml [hunk 1]
```diff
@@ -5483,8 +5485,8 @@
 		<info name="version" value="6.0" />
 		<info name="release" value="19931125"/>
 		<part name="flop1" interface="floppy_3_5">
-			<dataarea name="flop" size="1474560">
-				<rom name="mitsumi mouse driver 6.0.img" size="1474560" crc="591804ee" sha1="bb41f6447f631a4efe850431be3c720f9cadc4d1"/>
+			<dataarea name="flop" size="184320">
+				<rom name="mitsumi mouse driver 6.0.img" size="184320" crc="05bd4b0e" sha1="7a80b55a2d86a04aa19464b69cd0d524da7ef60c"/>
```
Comment (DopefishJustin) [inline hash/ibm5170.xml]: This is a 3.5" disk and according to the header it's 720KB. So the .img file ought to be 737280 bytes, even though it's not full.

Comment (FakeShemp) [reply hash/ibm5170.xml]: Is that a must with the img standard? The disk looks like this. There aren't even sectors above 180.
Mitsumi

Comment (DopefishJustin) [reply hash/ibm5170.xml]: Oh, weird!

The problem is, MAME won't load it like this:

```
Fatal error: Device 3.5" double density floppy drive load failed: Unable to iden
tify the image format
```

Comment (cracyc) [reply hash/ibm5170.xml]: Yup, it checks the image size and won't load it if it's not a known size.  I think most other emulators are like that too.  If you don't want to pad it then make it an image that has explicit sector numbering like IMD.

**Comment by Tafoid** [comment]
Finally got to checking these additions and we are missing one image (Commodore Amiga Hardware driver disks).  We have a kryoflux raw of that title, but no resulting .adf file.  Please upload this missing .adf

A590 Setup Disk 1.23 (Seagate Harddisk) [folder: a590_seagate - parent: a590 - size: 880kb]
missing set: A590 Setup Disk 1.23 (Seagate Harddisk)
missing rom: 317734-04.adf [size: 901120] [CRC32: 7f1623a7] [SHA1: 1f3e20206eb83f60745e19538de3123089c8da24]

**Comment by Tafoid** [comment]
Actually, cancel that.  I was able to find it was uploaded over a year ago and poked around a bit.
Thanks for the contributions!

**Pull request merged**